storage/imxrt-flash/flashsrv.c

```diff
@@ -94,100 +345,405 @@ static void flashsrv_devCtl(msg_t *msg)
+				mutexUnlock(memory->lock);
 				break;
 
 			default:
 				msg.o.io.err = -ENOSYS;
 				break;
 		}
-		msgRespond(port, &msg, rid);
+		msgRespond(memory->fOid.port, &msg, rid);
 	}
 }
 
 
-static uint32_t flashsrv_init(uint32_t *port)
+/* Initialization functions */
+
+static int flashsrv_initMeterfs(flashsrv_partition_t *part)
+{
+	meterfs_ctx_t *ctx;
+
+	if ((part->fsCtx = calloc(1, sizeof(meterfs_ctx_t))) == NULL) {
+		LOG_ERROR("imxrt-flashsrv: cannot allocate memory.");
+		return -ENOMEM;
+	}
+
+	ctx = (meterfs_ctx_t *)part->fsCtx;
+
+	ctx->sz = part->sz;
+	ctx->offset = part->offs;
+	ctx->sectorsz = flashsrv_common.flash_memories[part->fID].ctx.properties.sector_size;
+
+	if (part->fID == 0) {
+		ctx->read = flashsrv_mfsReadf0;
+		ctx->write = flashsrv_mfsWritef0;
+		ctx->partitionErase = flashsrv_mfsPartitionErasef0;
+		ctx->eraseSector = flashsrv_mfsEraseSectorf0;
+	}
+	else if (part->fID == 1) {
+		ctx->read = flashsrv_mfsReadf1;
+		ctx->write = flashsrv_mfsWritef1;
+		ctx->partitionErase = flashsrv_mfsPartitionErasef1;
+		ctx->eraseSector = flashsrv_mfsEraseSectorf1;
+	}
+
+	if (meterfs_init(ctx) < 0) {
+		LOG_ERROR("imxrt-flashsrv: init meterfs at flash: %d, partition: %d.", part->fID, part->oid.id);
+		return -1;
+	}
+
+	return 0;
+}
+
+
+static int flashsrv_mountPart(ptable_partition_t *pHeader, flashsrv_partition_t *part)
+{
+	int res = EOK;
+	char path[32];
+	void *mem;
+
+	sprintf(path, "/dev/flash%d.%s", part->fID, pHeader->name);
+
+	switch (pHeader->type) {
+		/* Raw partitions locate within single flash chip are handled by one thread */
+		case ptable_raw:
+			part->fsCtx = NULL;
+			part->oid.port = flashsrv_common.flash_memories[part->fID].rawPort;
+
+			if ((res = create_dev(&part->oid, path)) < 0) {
+				LOG_ERROR("imxrt-flashsrv: create %s - err: %d", path, res);
+				return res;
+			}
+
+			if (!flashsrv_common.flash_memories[part->fID].rawActive) {
+				flashsrv_common.flash_memories[part->fID].rawActive = 1;
+				if ((mem = malloc(THREAD_STACKSZ)) == NULL) {
+					LOG_ERROR("imxrt-flashsrv: cannot alloc memory.");
+					return -ENOMEM;
+				}
+
+				beginthread(flashsrv_rawThread, THREAD_PRIORITY, mem, THREAD_STACKSZ, (void *)&flashsrv_common.flash_memories[part->fID]);
+			}
+			part->pStatus = flash_memory_active;
+			break;
+
+		/* Each meterfs partiton is handled by separete thread */
+		case ptable_meterfs:
+			if ((res = flashsrv_initMeterfs(part)) < 0)
+				return res;
+
+			portCreate(&part->oid.port);
+			portRegister(part->oid.port, path, NULL);
+			if ((res = create_dev(&part->oid, path)) < 0) {
+				LOG_ERROR("imxrt-flashsrv: create %s - err: %d", path, res);
+				return res;
+			}
+
+			if ((mem = malloc(THREAD_STACKSZ)) == NULL) {
+				LOG_ERROR("imxrt-flashsrv: cannot alloc memory.");
+				return -ENOMEM;
+			}
+
+			beginthread(flashsrv_meterfsThread, THREAD_PRIORITY, mem, THREAD_STACKSZ, (void *)part);
+			part->pStatus = flash_memory_active;
+			break;
+
+		default:
+			break;
```

Comment:
FAIL: unknown FS type, return `EOK`

Reply:
I don't understand what you mean exactly.